addons/mail/static/src/core/web/message_patch.xml

```diff
@@ -5,23 +5,23 @@
             <t t-if="message.type === 'notification' or message.isTransient or message.trackingValues.length > 0" name="hasTrackingValue">
                 <t t-if="message.subtypeDescription">
                     <p class="mb-0">
-                        <t t-esc="message.subtypeDescription"/>
+                        <t t-out="searchHighlight(props.searchTerm, message.subtypeDescription)"/>
                     </p>
                 </t>
                 <t t-if="message.trackingValues.length">
                     <ul class="mb-0 ps-4">
                         <t name="trackingValues" t-foreach="message.trackingValues" t-as="trackingValue" t-key="trackingValue.id">
                             <li class="o-mail-Message-tracking mb-1" role="group">
-                                <span class="o-mail-Message-trackingOld me-1 px-1 text-muted fw-bold" t-esc="formatTrackingOrNone(trackingValue.oldValue)"/>
+                                <span class="o-mail-Message-trackingOld me-1 px-1 text-muted fw-bold" t-out="searchHighlight(props.searchTerm, formatTrackingOrNone(trackingValue.oldValue))"/>
                                 <i class="o-mail-Message-trackingSeparator fa fa-long-arrow-right mx-1 text-600"/>
-                                <span class="o-mail-Message-trackingNew me-1 fw-bold text-info" t-esc="formatTrackingOrNone(trackingValue.newValue)"/>
-                                <span class="o-mail-Message-trackingField ms-1 fst-italic text-muted">(<t t-esc="trackingValue.changedField"/>)</span>
+                                <span class="o-mail-Message-trackingNew me-1 fw-bold text-info" t-out="searchHighlight(props.searchTerm, formatTrackingOrNone(trackingValue.newValue))"/>
+                                <span class="o-mail-Message-trackingField ms-1 fst-italic text-muted">(<t t-out="searchHighlight(props.searchTerm, trackingValue.changedField)"/>)</span>
                             </li>
                         </t>
                     </ul>
                 </t>
                 <div t-if="message.body" class="o-mail-Message-body text-break mb-0 w-100" t-ref="body">
-                    <t t-out="message.body"/>
+                    <t t-out="searchHighlight(props.searchTerm, message.body)"/>
```

Comment:
I think the general idea of the function is good.
If you pass `messageSearch` as prop, then you don't need to define this util function, and instead everything related to search message feature is encapsulated in the `useMessageSearch()` hook.
```xml
<t t-out="props.messageSearch ? props.messageSearch.highlight(message.body) : message.body"/>
```

Might be a bit more readable with getters, and make all getters use `messageSearch.highlight()` conditionally when the feature is available.

Reply:
We can short it like this :) 
```javascript
props.messageSearch?.highlight(message.body) ?? message.body
```